lib/galaxy/tools/parameters/basic.py

```diff
@@ -2029,6 +2031,30 @@ def do_validate(v):
                 raise ValueError("At most %d datasets are required for %s" % (self.max, self.name))
 
 
+def src_id_to_item(
+    sa_session: "Session", value: Dict[str, Any], security: "IdEncodingHelper"
```

Comment:
Do you think you could work your type annotation magic and create overloads for value -> return type pairs ?